lms/djangoapps/verify_student/services.py

```diff
@@ -180,15 +167,12 @@ def user_status(cls, user):
 
         attempt = None
 
-        verifications = active_verifications(
-            cls.verifications_for_user(user),
-            timezone.now(),
-        )
+        verifications = cls.verifications_for_user(user)
 
         if verifications:
             attempt = verifications[0]
             for verification in verifications:
-                if verification.status == 'approved':
+                if verification.expiration_datetime > now() and verification.status == 'approved':
```

Comment:
I think there is something wrong with this logic. This loop goes through all the `active`, which means those attempts who are not expired. So your first condition `verification.expiration_datetime > now()` is always going to be `True`. All you need is to make sure the `verification.status == 'approved'`.
I think the best is to just revert this change and we should be OK.

Reply:
Oh, I see what's happening now. You actually retrieved all verifications. My bad. Now I think this is correct. Thanks for fixing that bug!